### DISCUSSION TITLE
How do I get categories sorted by alphabetical order?

### DISCUSSION THREAD

**Discussion by danmackinlay** [quarto-dev/quarto-cli#8225] [Q&A] [answered]
### Description

Currently my categories are not sorted alphabetically in English, in that if I have some categories that begin with a capital letter and some that begin with a lower-case letter, the ones beginning with a lower-case letter always come later, regardless of whether this makes sense for the language they are in. However, English speakers (and AFAIK most speakers of languages using the roman alphabet) typically expect that "A" and "a" should be sorted next to each other, i.e. capitalisation is not considered important when sorting in alphabetical order.

For an example, see the categories sidebar of [my blog](https://danmackinlay.name/notebook):
<img width="219" alt="Screenshot 2024-01-11 at 17 44 57" src="https://github.com/quarto-dev/quarto-cli/assets/21740/13572f8f-5a7b-4360-9c8c-772a0a33844f">

It looks weird that words like "academe" seem to come alphabetically later than "Markov"

Does quarto support alphabetical sorting?

**Comment by mcanouil** [comment]
This is currently not possible as the category sorting is done internally in TypeScript: https://github.com/quarto-dev/quarto-cli/blob/main/src/project/types/website/listing/website-listing-categories.ts#L74-L80

It would mean to have something like below instead:

```ts
  function compareCategories(a: string, b: string) {
    return a.localeCompare(b, undefined, { sensitivity: "base" });
  }

  for (const cat of Object.keys(cats).sort(compareCategories)) {
    const count = cats[cat];
    const catEl = categoryElement(doc, cat, formatFn(cat, count), cat);
    categoriesEl.appendChild(catEl);
  }
```

<img width="99" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/4cc67ce5-0d5e-4669-9775-1d2dbc617427">


cc @dragonstyle do you think it's worth adding? I am no expert in TypeScript, so might not be very optimised (as in the best function for the job) but does the job.

**Reply by dragonstyle** [reply]
This definitely makes sense to do and is very easy.

**Reply by batpigandme** [reply]
Is there an issue I can track for this? No rush—just want to make sure I catch it when it comes out! 🙂

**Reply by mcanouil** [reply, marked as answer]
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/cf8a58f8-c61b-49eb-9a24-aa091f31debd)

- https://github.com/quarto-dev/quarto-cli/issues/8238

**Reply by mcanouil** [reply]
GitHub UI/UX is bad ...

![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/b9659403-5876-47e8-a054-d0eb2fd47ed5)

1m => 1 minute ago
2m => 2 months ago

**Reply by batpigandme** [reply]
Ah! Thank you. Sorry, I'm new to the discussions interface!

**Reply by mcanouil** [reply]
well, as you can see, the interface is not great ^^
